### DISCUSSION TITLE
What will the upgrade path from 2.x to 3.x look like ?

### DISCUSSION THREAD

**Discussion by OlivierFortier** [requarks/wiki#6191] [Help / Questions]
I am currently looking at wiki.js to start a new project but I am afraid that I won't be able to easily upgrade to 3.x once it releases.

Will I be able to reuse my custom modules and plugins from 2.x in 3.x ? 
Will I be able to keep my content and media architecture ? 
Will I be able to keep my pages ? 
Will I be able to keep my auth ?

I have tried looking for this information online but I have not had any luck finding it.

**Comment by cdn0x12** [comment]
You can refer to:
https://blog.js.wiki/news/2022/the-road-to-the-wiki-js-3-beta

3.x is still under development. It stands to reason that the full migration guide will only be available after the official release. While you can easily export pages, assets, comments, users, etc.using the export utility, the  custom module may need to be migrated manually.
